foundry/foundry.py

```diff
@@ -747,7 +708,7 @@ def _load_data(self, file=None, source_id=None, as_hdf5=False):
             try:
                 path_to_file = os.path.join(path, file)
             except Exception as e:
-                print(f"Unable to find path to file for download: {e}")
+                logging.log(logging.FATAL, f"Unable to find path to file for download: {e}")  # is this even necessary?
```

Comment:
Hm it looks like there is some conflicting comments/intentions here. `os.path.join()` will rarely error out afaik. 

My suggestion would be to remove this try/except block, unless @kjschmidt913 has any reasons we shouldn't. The important thing is to check that the file exists, which is done in the next block

Reply:
☝🏼 Ari's on point. `join` will  fail if file or path is `None` but that's a roundabout why of testing what we're looking for. We should directly test those conditions rather than rely on a secondary mechanism.

Plus, it is better to contain the error message in the exception (e.g., `raise ValueError('Unable to find path...')`)

Putting it together:

```python

if path is None or file is None:
    raise ValueError(........)
path_to_file = os.path.join(path, file)
```

Reply:
@Aadit-Ambadkar , you still haven't addressed this comment about the use of a try/except block when an if statement is clearer.

I don't think we need to rely on the FATAL to notify people when also throwing an exception.

Reply:
I think what was intended was os.path.exists(os.path.join(...,...))

Reply:
@Aadit-Ambadkar could you please address this block as well.